File: src/main/java/com/pedigree/render/NodeMetrics.java
package com.pedigree.render;

/**
 * Provides default node metrics; can be replaced with text-aware sizing.
 */
public class NodeMetrics {
    private final double defaultWidth;
    private final double defaultHeight;

    public NodeMetrics() {
        this(120.0, 60.0);
    }

    public NodeMetrics(double defaultWidth, double defaultHeight) {
        this.defaultWidth = defaultWidth;
        this.defaultHeight = defaultHeight;
    }

    public double getWidth(String nodeId) {
        return defaultWidth;
    }

    public double getHeight(String nodeId) {
        return defaultHeight;
    }
}

File: src/main/kotlin/com/family/tree/render/NodeMetrics.kt
package com.family.tree.render

/**
 * Provides default node metrics; can be replaced with text-aware sizing.
 */
open class NodeMetrics(
    private val defaultWidth: Double = 120.0,
    private val defaultHeight: Double = 60.0
) {
    open fun getWidth(nodeId: String): Double = defaultWidth
    open fun getHeight(nodeId: String): Double = defaultHeight
}
